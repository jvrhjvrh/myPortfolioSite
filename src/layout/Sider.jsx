import React, { useState } from 'react';
import { Layout, Menu } from 'antd';
import { ExperimentOutlined, HomeOutlined, AppstoreOutlined } from '@ant-design/icons';
import propTypes from 'proptypes';
import { useHistory } from 'react-router-dom';

const { Content, Sider } = Layout;

const SiderComponent = ({ children }) => {
    const [collapsed, setCollapsed] = useState(false);
    const history = useHistory();

    const onCollapse = (collapse) => {
        setCollapsed(collapse);
    };

    const onSelectMenu = ({ key }) => {
        switch (key) {
            case '2':
                history.push('/brainfuck');
                break;
            case '3':
                history.push('/pixelart');
                break;
            default:
                history.push('/');
        }
    };

    return (
        <Layout style={{ height: '100vh' }}>
            <Sider collapsed={collapsed} collapsible onCollapse={onCollapse}>
                <Menu theme="dark" mode="inline" onClick={onSelectMenu}>
                    <Menu.Item key="1">
                        <HomeOutlined />
                        <span>home.js</span>
                    </Menu.Item>
                    <Menu.Item key="2">
                        <ExperimentOutlined />
                        <span>Brainfuck.js</span>
                    </Menu.Item>
                    <Menu.Item key="3">
                        <AppstoreOutlined />
                        <span>Pixelart.js</span>
                    </Menu.Item>
                </Menu>
            </Sider>
            <Layout>
                <Content style={{ margin: '0 16px' }}>
                    {children}
                </Content>
            </Layout>
        </Layout>
    );
};

SiderComponent.propTypes = {
    children: propTypes.node.isRequired,
};

export default SiderComponent;
