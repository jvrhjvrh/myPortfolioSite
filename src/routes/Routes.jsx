import React from 'react';
import {
    Switch,
    Route,
} from 'react-router-dom';
import Home from '../pages/home';
import Brainfuck from '../pages/brainfuck';

export default () => (
    <Switch>
        <Route path="/brainfuck" component={Brainfuck} />
        <Route path="/" component={Home} />
    </Switch>
);
