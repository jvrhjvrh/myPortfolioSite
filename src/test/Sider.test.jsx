import React from 'react';
import { BrowserRouter, Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { render, fireEvent } from '@testing-library/react';
import Sider from '../layout/Sider';
import Routes from '../routes/Routes';

const PropChildren = () => (
    <div> teste </div>
);

it('render correctly', () => {
    const component = render(
        <div data-testid="custom-element">
            <BrowserRouter>
                <Sider><PropChildren /></Sider>
            </BrowserRouter>
        </div>,
    );
    expect(component.getByTestId('custom-element')).toMatchSnapshot();
});

it('collapses correctly', () => {
    const component = render(
        <BrowserRouter>
            <Sider><PropChildren /></Sider>
        </BrowserRouter>,
    );
    const button = component.getByLabelText('left');
    fireEvent.click(button);
    component.getByLabelText('right');
});

it('changes to brainfuck page', () => {
    const history = createMemoryHistory();
    const component = render(
        <Router history={history}>
            <Sider>
                <Routes />
            </Sider>
        </Router>,
    );

    component.getByTestId('home-page');

    const button = component.getByText('Brainfuck.js');

    fireEvent.click(button);

    component.getByTestId('brainfuck-page');
});

it('changes to brainfuck page and back to home', () => {
    const history = createMemoryHistory();
    const component = render(
        <Router history={history}>
            <Sider>
                <Routes />
            </Sider>
        </Router>,
    );

    component.getByTestId('home-page');

    const button = component.getByText('Brainfuck.js');

    fireEvent.click(button);

    component.getByTestId('brainfuck-page');

    const homeButton = component.getByText('home.js');

    fireEvent.click(homeButton);

    component.getByTestId('home-page');
});

it('changes to pixelart page', () => {
    const history = createMemoryHistory();
    const component = render(
        <Router history={history}>
            <Sider>
                <Routes />
            </Sider>
        </Router>,
    );

    component.getByTestId('home-page');

    const button = component.getByText('Pixelart.js');

    fireEvent.click(button);

    component.getByTestId('pixelart-page');
});
