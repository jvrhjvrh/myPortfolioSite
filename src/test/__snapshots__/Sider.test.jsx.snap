// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<div
  data-testid="custom-element"
>
  <section
    class="ant-layout ant-layout-has-sider"
    style="height: 100vh;"
  >
    <aside
      class="ant-layout-sider ant-layout-sider-dark ant-layout-sider-has-trigger"
      style="flex: 0 0 200px; max-width: 200px; min-width: 200px; width: 200px;"
    >
      <div
        class="ant-layout-sider-children"
      >
        <ul
          class="ant-menu ant-menu-dark ant-menu-root ant-menu-inline"
          role="menu"
        >
          <li
            class="ant-menu-item"
            role="menuitem"
            style="padding-left: 24px;"
          >
            <span
              aria-label="home"
              class="anticon anticon-home"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="home"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
                />
              </svg>
            </span>
            <span>
              home.js
            </span>
          </li>
          <li
            class="ant-menu-item"
            role="menuitem"
            style="padding-left: 24px;"
          >
            <span
              aria-label="experiment"
              class="anticon anticon-experiment"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="experiment"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M512 472a40 40 0 1080 0 40 40 0 10-80 0zm367 352.9L696.3 352V178H768v-68H256v68h71.7v174L145 824.9c-2.8 7.4-4.3 15.2-4.3 23.1 0 35.3 28.7 64 64 64h614.6c7.9 0 15.7-1.5 23.1-4.3 33-12.7 49.4-49.8 36.6-82.8zM395.7 364.7V180h232.6v184.7L719.2 600c-20.7-5.3-42.1-8-63.9-8-61.2 0-119.2 21.5-165.3 60a188.78 188.78 0 01-121.3 43.9c-32.7 0-64.1-8.3-91.8-23.7l118.8-307.5zM210.5 844l41.7-107.8c35.7 18.1 75.4 27.8 116.6 27.8 61.2 0 119.2-21.5 165.3-60 33.9-28.2 76.3-43.9 121.3-43.9 35 0 68.4 9.5 97.6 27.1L813.5 844h-603z"
                />
              </svg>
            </span>
            <span>
              Brainfuck.js
            </span>
          </li>
          <li
            class="ant-menu-item"
            role="menuitem"
            style="padding-left: 24px;"
          >
            <span
              aria-label="appstore"
              class="anticon anticon-appstore"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="appstore"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M464 144H160c-8.8 0-16 7.2-16 16v304c0 8.8 7.2 16 16 16h304c8.8 0 16-7.2 16-16V160c0-8.8-7.2-16-16-16zm-52 268H212V212h200v200zm452-268H560c-8.8 0-16 7.2-16 16v304c0 8.8 7.2 16 16 16h304c8.8 0 16-7.2 16-16V160c0-8.8-7.2-16-16-16zm-52 268H612V212h200v200zM464 544H160c-8.8 0-16 7.2-16 16v304c0 8.8 7.2 16 16 16h304c8.8 0 16-7.2 16-16V560c0-8.8-7.2-16-16-16zm-52 268H212V612h200v200zm452-268H560c-8.8 0-16 7.2-16 16v304c0 8.8 7.2 16 16 16h304c8.8 0 16-7.2 16-16V560c0-8.8-7.2-16-16-16zm-52 268H612V612h200v200z"
                />
              </svg>
            </span>
            <span>
              Pixelart.js
            </span>
          </li>
        </ul>
      </div>
      <div
        class="ant-layout-sider-trigger"
        style="width: 200px;"
      >
        <span
          aria-label="left"
          class="anticon anticon-left"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="left"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
            />
          </svg>
        </span>
      </div>
    </aside>
    <section
      class="ant-layout"
    >
      <main
        class="ant-layout-content"
        style="margin: 0px 16px;"
      >
        <div>
           teste 
        </div>
      </main>
    </section>
  </section>
</div>
`;
